sd.go

```diff
@@ -73,6 +78,38 @@ func LookupSidByName(name string) (sid string, err error) {
 	return sid, nil
 }
 
+// LookupNameBySid looks up the name of an account by SID
+func LookupNameBySid(sid string) (name string, err error) {
+	if sid == "" {
+		return "", &AccountLookupError{sid, cERROR_NONE_MAPPED}
+	}
+
+	sidBuffer, err := syscall.UTF16FromString(sid)
+	if err != nil {
+		return "", &AccountLookupError{sid, err}
+	}
+
+	var sidPtr *byte
+	convertStringSidToSid(&sidBuffer[0], &sidPtr)
```

Comment:
Is the pointer to the first elem needed? I'd think you could just pass sidBuffer directly. Also seems like this function can fail so can we add an (err error) to the //sys definition? https://github.com/golang/sys/blob/1c1b9b1eba6a51cf62024505e66017fcd6aa559f/windows/security_windows.go#L187

Reply:
`sidBuffer` is a []uint16 and can't be passed to `convertStringSidToSid` directly. I have added an error check, though.

Reply:
Ahh, I'm so used to the windows.UTF16PtrFromString variant I didn't even notice

Reply:
I had to double-check as well. All those pointer-slice conversions can get pretty confusing. :smiley:

Reply:
It'd be nice to swap to that function but not a blocker

Reply:
OK, I've just done that.